compiler/z/codegen/S390GenerateInstructions.cpp

```diff
@@ -361,6 +361,20 @@ generateS390CompareAndBranchInstruction(TR::CodeGenerator * cg,
    return returnInstruction;
    }
 
+TR::Instruction *
+generateS390CompareAndBranchInstruction(TR::CodeGenerator * cg,
+                                        TR::InstOpCode::Mnemonic compareOpCode,
+                                        TR::Node * node,
+                                        TR::Register * first ,
+                                        TR::RealRegister * second,
```

Comment:
Same concern here as @r30shah. It seems odd that we would be passing in real registers here. It must mean we are using this API after RA. I would assume `first` should also be a `TR::RealRegister` then, otherwise the API doesn't make sense.

Reply:
@fjeremic  Yes, In the testing @simonameng saw cases where we call this API with `TR::RealRegister` . I suspect originally it was casting it to `TR::Register` implicitly but now as we have used template to represent type of `second` compiler was looking for `generateS390CompareAndBranchInstruction<TR::RealRegister*>` .  Though agree with you, `first` should also be `TR::RealRegister` in cases when `second` is.

Reply:
I add the version of `TR::RealRegister` because I receive a compilation error saying that in some use cases, there are some undefined symbol having `TR::RealRegister` type argument for `second`.

Reply:
The error message said the undefined symbol is the signature with `TR::Register first` and `TR::RealRegister second`.

Reply:
@r30shah Those are the two use cases having `TR::RealRegister` type argument:

https://github.com/eclipse/openj9/blob/81195db9cc55be1cd09f9534706823f409b9aa68/runtime/compiler/z/codegen/ReduceSynchronizedFieldLoad.cpp#L224

https://github.com/eclipse/openj9/blob/81195db9cc55be1cd09f9534706823f409b9aa68/runtime/compiler/z/codegen/ReduceSynchronizedFieldLoad.cpp#L232

Reply:
Good one @simonameng  to figure out the use cases where we have passed in `TR::RealRegister` to this API. As the use cases are in `OpenJ9` repository I suggest you to add another commit to [this OpenJ9 PR](https://github.com/eclipse/openj9/pull/6725) of yours which makes changes in use cases of the API modified in this PR.

Basically you need a changes in `ReduceSynchronizedFieldLoad::inlineSynchronizedFieldLoad` that declares a variable of type `TR::Register` which receives return value from the codegen query `getMethodMetaDataRealRegister` and use that to pass to this API instead of passing it what we receive from the query directly to API
This way we can eliminate the need of any another function that increases complexity of this API unnecessarily.